modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -641,18 +641,20 @@ void ONNXImporter::handleNode(const opencv_onnx::NodeProto& node_proto_)
             int axis = 0;
             std::vector<int> begin;
             std::vector<int> end;
+            std::vector<int> steps;
             int inp_size = node_proto.input_size();
 
             if (inp_size == 1)
             {
                 if (layerParams.has("steps"))
                 {
-                    DictValue steps = layerParams.get("steps");
-                    for (int i = 0; i < steps.size(); ++i)
+                    DictValue steps_dict = layerParams.get("steps");
+                    for (int i = 0; i < steps_dict.size(); ++i)
                     {
-                        if (steps.get<int>(i) != 1)
+                        if (steps_dict.get<int>(i) != 1) {
                             CV_Error(Error::StsNotImplemented,
                                 "Slice layer only supports steps = 1");
```

Comment:
Do we want to support this case?

Reply:
@alalek, this case could be useful for opset9, which doesn't support step!=1, however, it means that initially onnx model could not be generated with step value != 1, hence, this check is useless.
Just in case, I've checked the model from the [issue](https://github.com/opencv/opencv/issues/14672) (which relates to the PR, where this code block was added), there is no step parameter (in opset9 slice layer no step parameter => step=1),  so I'll remove this code.